src/datasources/email/email.datasource.spec.ts

```diff
@@ -187,4 +188,73 @@ describe('Email Datasource Tests', () => {
       }),
     ).rejects.toThrow(EmailAddressDoesNotExistError);
   });
+
+  it('gets only verified email addresses associated with a given safe address', async () => {
+    const chainId = faker.number.int({ max: DB_CHAIN_ID_MAX_VALUE }).toString();
+    const safeAddress = faker.finance.ethereumAddress();
+    const verifiedSigners = [
+      {
+        emailAddress: faker.internet.email(),
+        signer: faker.finance.ethereumAddress(),
+        code: faker.number.int({ max: 999998 }).toString(),
+      },
+      {
+        emailAddress: faker.internet.email(),
+        signer: faker.finance.ethereumAddress(),
+        code: faker.number.int({ max: 999998 }).toString(),
+      },
+    ];
+    const nonVerifiedSigners = [
+      {
+        emailAddress: faker.internet.email(),
+        signer: faker.finance.ethereumAddress(),
+        code: faker.number.int({ max: 999998 }).toString(),
+      },
+      {
+        emailAddress: faker.internet.email(),
+        signer: faker.finance.ethereumAddress(),
+        code: faker.number.int({ max: 999998 }).toString(),
+      },
+    ];
+    await Promise.all(
+      verifiedSigners.map(async ({ emailAddress, signer, code }) => {
+        await target.saveEmail({
+          chainId,
+          safeAddress,
+          emailAddress,
+          signer,
+          code,
+        });
+        await target.verifyEmail({
+          chainId: chainId,
+          safeAddress,
+          signer,
+        });
+      }),
+    );
+    await Promise.all(
+      nonVerifiedSigners.map(async ({ emailAddress, signer, code }) => {
+        await target.saveEmail({
+          chainId,
+          safeAddress,
+          emailAddress,
+          signer,
+          code,
+        });
+      }),
+    );
+
+    const result = await target.getVerifiedSignerEmailsBySafeAddress({
+      chainId,
+      safeAddress,
+    });
+
+    const expected = orderBy(
+      verifiedSigners.map((verifiedSigner) => ({
+        email: verifiedSigner.emailAddress,
+      })),
+      'email',
+    );
```

Comment:
This `orderBy` email address is not guaranteed by the database correct? If we want to be deterministic in the return then this should be covered either by the domain layer or the datasource itself.

Reply:
I've added an `ORDER BY` clause to the database query to be deterministic. I think it's consistent to order by the primary key (insertion order) but I'm open to other options.

Added in [f1ff610](https://github.com/safe-global/safe-client-gateway/pull/877/commits/f1ff61004d4b054ac07ac7b5750cec600b9d5d8a)